datajunction-server/tests/api/dimension_links_test.py

```diff
@@ -503,7 +503,7 @@ async def test_link_complex_dimension_with_role(
 FROM default_DOT_events
 LEFT JOIN default_DOT_users
   ON default_DOT_events.user_id = default_DOT_users.user_id
-  AND default_DOT_events.event_start_date BETWEEN default_DOT_users.snapshot_date AND CAST(DATE_ADD(CAST(default_DOT_users.snapshot_date AS DATE), 10) AS INT)
+  AND default_DOT_events.event_start_date = default_DOT_users.snapshot_date
```

Comment:
Any ideas why did this unit test changed?

Reply:
This is most likely from the join order changing for dimensions discovery, but also the old unit tests don't look quite right either